src/Framework/ClassResolver/AbstractClassResolver.php

```diff
@@ -156,4 +171,17 @@ private function createDefaultGacelaClass(): ?object
                 return null;
         }
     }
+
+    private function triggerEvent(GacelaEventInterface $event): void
```

Comment:
I think it makes sense to move this logic to a new class with a `handle/trigger` method/whatsoever.
This way, we are not coupled to the `doResolve()` and we can add any other listener in the application, what do you think?